compiler/noirc_evaluator/src/ssa/function_builder/data_bus.rs

```diff
@@ -142,7 +142,12 @@ impl FunctionBuilder {
         assert!(databus.databus.is_none(), "initializing finalized call data");
         let typ = self.current_function.dfg[value].get_type().into_owned();
         match typ {
-            Type::Numeric(_) => {
+            Type::Numeric(numeric_type) => {
+                let value = if matches!(numeric_type, NumericType::NativeField) {
```

Comment:
nit: Simplification should handle this as well.

Reply:
Which one?

Reply:
In `simplify_cast` if we go Field -> Field we just return the value